__test__/unit/blockchain_api/error_scenario_runner.ts

```diff
@@ -51,38 +69,159 @@ export class ErrorScenarioRunner {
+
+        this.simpleInterestLoan = new SimpleInterestLoanAdapter(this.web3, this.contractsAPI);
 
         // Mark instance as configured.
         this.isConfigured = true;
     }
 
+    public testRepaymentRouterErrorScenario(scenario: RepaymentRouterErrorScenario) {
+        describe(scenario.description, () => {
+            let txHash: string;
+
+            const AMOUNT = Units.ether(10);
+
+            beforeEach(async () => {
+                const principalTokenAddress = await this.tokenRegistry.getTokenAddressBySymbol.callAsync(
+                    "REP",
+                );
+
+                const principalToken = await DummyTokenContract.at(
+                    principalTokenAddress,
+                    this.web3,
+                    TX_DEFAULTS,
+                );
+
+                // Grant creditor a balance of tokens
+                await principalToken.setBalance.sendTransactionAsync(CREDITOR, AMOUNT, {
+                    from: CONTRACT_OWNER,
+                });
+
+                // Grant debtor a balance of tokens
+                await principalToken.setBalance.sendTransactionAsync(DEBTOR, AMOUNT, {
+                    from: CONTRACT_OWNER,
+                });
+
+                // Approve the token transfer proxy for a sufficient
+                // amount of tokens for an order fill.
+                await principalToken.approve.sendTransactionAsync(
+                    this.tokenTransferProxy.address,
+                    AMOUNT,
+                    { from: CREDITOR },
+                );
+
+                const debtOrder = await this.simpleInterestLoan.toDebtOrder({
+                    debtor: DEBTOR,
+                    creditor: CREDITOR,
+                    principalAmount: AMOUNT,
+                    principalToken: principalToken.address,
+                    interestRate: new BigNumber(0.1),
+                    amortizationUnit: "months",
+                    termLength: new BigNumber(2),
+                    salt: new BigNumber(0),
+                });
+
+                debtOrder.debtorSignature = await this.signerAPI.asDebtor(debtOrder);
+
+                const debtOrderWrapped = await DebtOrderWrapper.applyNetworkDefaults(
+                    debtOrder,
+                    this.contractsAPI,
+                );
+
+                const issuanceHash = debtOrderWrapped.getIssuanceCommitmentHash();
+
+                if (scenario.agreementExists) {
+                    await this.orderAPI.fillAsync(debtOrder, { from: CREDITOR });
+                }
+
+                if (scenario.isPayerBalanceInsufficient) {
+                    await principalToken.setBalance.sendTransactionAsync(DEBTOR, ZERO_AMOUNT, {
+                        from: CONTRACT_OWNER,
+                    });
+                }
+
+                let repaymentToken: DummyTokenContract;
+
+                if (scenario.isRouterAuthorizedToReportPayment) {
+                    repaymentToken = principalToken;
```

Comment:
I'm okay with triggering the router authorization error by choosing a _specific_ instance of how this can happen (i.e.  when repayment is made for `SimpleInterestTermsContract` in unexpected token), but let's be explicit about this assumption in a comment.  Could help save a lot of debugging time if ever in the future we change the logic of how `SimpleInterestTermsContract` works.

Reply:
I've made the description related to this scenario more explicit. let me know what you think.